tests/by-util/test_timeout.rs

```diff
@@ -143,18 +143,29 @@ fn test_kill_after_long() {
         .no_stderr();
 }
 
+/// Test that a command that is executed in a sub-shell is killed/terminated.
 #[test]
-fn test_kill_subprocess() {
+fn test_kill_subshell() {
     new_ucmd!()
         .args(&[
-            // Make sure the CI can spawn the subprocess.
-            "10",
+            "1",
+            // Spawn command in sub-shell.
             "sh",
             "-c",
-            "sh -c \"trap 'echo xyz' TERM; sleep 30\"",
+            // Set a trap to print 'xyz' when the sub-shell receives a
+            // TERM signal.
+            "trap 'echo xyz' TERM; sleep infinity",
```

Comment:
Is `sleep infinity` allowed/an anti-pattern? While it tests a correct SUT well (the sub-shell command should, if the implementation is correct, be interrupted/woken after the timeout), it would be problematic for testing an incorrect SUT (if the `TERM` signal is never sent, this test executes forever). Therefore, should I use some magic number here? Maybe `10`?

Reply:
I believe the test should be killed by the test framework automatically, but a magic number sounds good as well.

Reply:
I would personally prefer the `sleep infinity` (it's more obvious that the only way this terminates is if sent `TERM`), and would only use the magic number if you prefer that.